galleries/examples/images_contours_and_fields/multi_image.py

```diff
@@ -12,47 +18,54 @@
 from matplotlib import colors
 
 np.random.seed(19680801)
-Nr = 3
-Nc = 2
 
-fig, axs = plt.subplots(Nr, Nc)
+datasets = [
+    (i+1)/10 * np.random.rand(10, 20)
+    for i in range(4)
+]
+
+fig, axs = plt.subplots(2, 2)
 fig.suptitle('Multiple images')
 
-images = []
-for i in range(Nr):
-    for j in range(Nc):
-        # Generate data with a range that varies from one plot to the next.
-        data = ((1 + i + j) / 10) * np.random.rand(10, 20)
-        images.append(axs[i, j].imshow(data))
-        axs[i, j].label_outer()
+# create a single norm to be shared across all images
+norm = colors.Normalize(vmin=np.min(datasets), vmax=np.max(datasets))
 
-# Find the min and max of all colors for use in setting the color scale.
-vmin = min(image.get_array().min() for image in images)
-vmax = max(image.get_array().max() for image in images)
-norm = colors.Normalize(vmin=vmin, vmax=vmax)
-for im in images:
-    im.set_norm(norm)
+images = []
+for ax, data in zip(axs.flat, datasets):
+    images.append(ax.imshow(data, norm=norm))
 
 fig.colorbar(images[0], ax=axs, orientation='horizontal', fraction=.1)
 
-
-# Make images respond to changes in the norm of other images (e.g. via the
-# "edit axis, curves and images parameters" GUI on Qt), but be careful not to
-# recurse infinitely!
-def update(changed_image):
-    for im in images:
-        if (changed_image.get_cmap() != im.get_cmap()
-                or changed_image.get_clim() != im.get_clim()):
-            im.set_cmap(changed_image.get_cmap())
-            im.set_clim(changed_image.get_clim())
-
-
-for im in images:
-    im.callbacks.connect('changed', update)
-
 plt.show()
 
 # %%
+# The colors are now kept consistent across all images when changing the
+# scaling, e.g. through zooming in the colorbar or via the "edit axis,
+# curves and images parameters" GUI of the Qt backend. This is sufficient
+# for most practical use cases.
+#
+# Advanced: Additionally sync the colormap
+# ----------------------------------------
+#
+# While the norm is shared, the colormaps are not. This is often ok
```

Comment:
Does the new wording clarify things?

> While the norm is shared in the above example, the colormaps are not:
> Each image holds an independent reference to the colormap it wants to
> use. We implicitly rely on all images using the same colormap. This is
> often ok because colormaps are usually not changed dynamically.